mapscript/php/php_mapscript.c

```diff
@@ -898,9 +898,7 @@ PHP_FUNCTION(ms_tokenizeMap)
                                         filename);
     return;
   } else {
-    if (array_init(return_value) == FAILURE) {
-      RETURN_FALSE;
-    }
+    array_init(return_value);
```

Comment:
Can this still return a FAILURE? What is the main reason for changing these?

Reply:
From PHP 7.3 onwards `array_init()` is no longer a function and no longer returns a value (see [UPGRADING.INTERNALS](https://github.com/php/php-src/blob/PHP-7.3/UPGRADING.INTERNALS#L44)) and so from PHP 7.3 onwards cannot be used in the if-condition.

In any case the return value was not checked in every case in the code where `array_init()` is used, and as far as I can see then at least as far back as PHP 5.6 it was impossible for  `array_init()` to return anything other than `SUCCESS`.

What happens in case of failure is a good question - but I assume that this would be pretty fatal in any case (out of memory? I can't think of any other likely error causes) and so (i) there was never much point checking the return value and (ii) that would be the reason why there is no longer a return value.

I guess it would be possible to shim all the uses of array_init() such that up until 7.2 the return value is checked, but as far as I can see that would be pretty pointless, so it's simpler, cleaner and has no disadvantage to just use it in the "new" style without the return check.

Reply:
@ejn - thanks for the thorough explanation.